templates/index.md.tmpl

```diff
@@ -26,12 +39,39 @@ In future the provider will be extended to support more of Juju's capabilities.
 
 The provider has a dependency on Juju CLI configuration store.
 
-In order to operate, the provider expects configuration to be found in one of the following locations:
+## Authentication
+
+There are 3 ways to define credentials for authentication with the juju controller you wish to use. They are displayed in order in which the provider looks for credentials.
+
+### Static credentials
+
+Define the juju controller credentials in the provider definition in your terraform plan.
+
+``` terraform
+provider "juju" {
+  controller_addresses = "10.225.205.241:17070,10.225.205.242:17070"
+  username = "jujuuser"
+  password = "password1"
+  ca_certificate = file("~/ca-cert.pem")
+}
+```
+
+### Environment variables
+
+Define the juju controller credentials in the provider definition via environment variables. These can be setup as follows:
+
+```shell
+export CONTROLLER=$(juju whoami | yq .Controller)
+export JUJU_CONTROLLER_ADDRESSES="$(juju show-controller | yq '.[$CONTROLLER]'.details.\"api-endpoints\" | tr -d "[]' "|tr -d '"'|tr -d '\n')"
+export JUJU_USERNAME="$(cat ~/.local/share/juju/accounts.yaml | yq .controllers.$CONTROLLER.user|tr -d '"')"
+export JUJU_PASSWORD="$(cat ~/.local/share/juju/accounts.yaml | yq .controllers.$CONTROLLER.password|tr -d '"')"
+export JUJU_CA_CERT="$(juju show-controller $(echo $CONTROLLER|tr -d '"') | yq '.[$CONTROLLER]'.details.\"ca-cert\"|tr -d '"'|sed 's/\\n/\n/g')"
+```
 
-* `$XDG_DATA_HOME/juju`
-* `~/.local/share/juju`
+### Populated by provider via the juju cli client.
```

Comment:
It might be worth to also add here that the provider is not dependent on the juju-cli (mostly to avoid future queries about that).

Reply:
See my comments for the other file.